src/index.ts

```diff
@@ -5,25 +5,26 @@ type TagNode = ReturnType<typeof parser>[number]
 type ReactElements = Record<string,ReactNode | ((children?: ReactNode) => ReactNode)>
 interface ParserOptions {strict?: boolean, tag?: [string, string]}
 
-export const interpolate = createInterpolate()
-
-export function createInterpolate (opts: ParserOptions = {}) {
-    const parse = (text: string) => parser(text, opts)
-    return (text: string, elements: ReactElements) => ensureOneNode(nodesToReactNodes(parse(text), elements))
+const parserWithOpts = (opts: ParserOptions = {}, text: string) => parser(text, opts)
+const parserWithOptsAndMemo = (opts: ParserOptions = {}, text: string) => useMemo(() => parserWithOpts(opts, text), [text])
+const parse = (parser: Function) => (opts: ParserOptions = {}) => {
```

Comment:
parse란 이름대신 다른 이름을 지어줄순 없을까요
parse는 가장 마지막에 리턴되는 함수에 어울리는 이름인 것 같습니다.

Reply:
parse를 createParser로 변경하는 것은 어떨까요?
```suggestion
const createParser = (parser: Function) => (opts: ParserOptions = {}) => {
```

Reply:
```suggestion
const createParserCreator = (parser: Function) => (opts: ParserOptions = {}) => {
```
음 커링이 두번되고 있기때문에 createParserCreator로 하겠습니담 😃